crypto/rand/rand_lib.c

```diff
@@ -25,6 +25,45 @@ static CRYPTO_RWLOCK *rand_meth_lock;
 static const RAND_METHOD *default_RAND_meth;
 static CRYPTO_ONCE rand_init = CRYPTO_ONCE_STATIC_INIT;
 
+#ifdef OPENSSL_RAND_SEED_RDTSC
+/*
+ * Since we get some randomness from the low-order bits of the
+ * high-speec clock, it can help.  But don't return a status since
+ * it's not sufficient to indicate whether or not the seeding was
+ * done.
+ */
+void rand_rdtsc(void)
+{
+    unsigned char c;
+    int i;
+
+    for (i = 0; i < 10; i++) {
+        c = (unsigned char)(OPENSSL_rdtsc() & 0xFF);
```

Comment:
There is some randomness in the higher order bits which would be a shame to throw away.  Maybe something like:

```
unsigned int x = OPENSSL_rdtsc();
x ^= x >> 16;
x ^= x >> 8;
c = x & 255;
```

If we knew `OPENSSL_rdtsc` was returning a `uint64_t` the first reduction step could be `x ^= x >> 32;`  Not a biggie, there isn't much randomness in the high bits.